pkg/scalers/influxdb_scaler.go

```diff
@@ -0,0 +1,187 @@
+	}
+
+	if val, ok := config.TriggerMetadata["thresholdValue"]; ok {
+		value, err := strconv.ParseFloat(val, 10)
+		if err != nil {
+			return nil, fmt.Errorf("thresholdValue: failed to parse thresholdValue length %s", err.Error())
+		}
+		thresholdValue = float64(value)
+	} else {
+		return nil, fmt.Errorf("no threshold value given")
+	}
+
+	return &influxDBMetadata{
+		authToken:        authToken,
+		organizationName: organizationName,
+		query:            query,
+		serverURL:        serverURL,
+		thresholdValue:   thresholdValue,
+	}, nil
+}
+
+// IsActive returns true if queried value is above the minimum value
+func (s *influxDBScaler) IsActive(ctx context.Context) (bool, error) {
+	queryAPI := s.client.QueryAPI(s.metadata.organizationName)
+
+	value, err := queryInfluxDB(queryAPI, s.metadata.query)
+	if err != nil {
+		return false, err
+	}
+
+	return value > s.metadata.thresholdValue, nil
+}
+
+// Close closes the connection of the client to the server
+func (s *influxDBScaler) Close() error {
+	s.client.Close()
+	return nil
+}
+
+// queryInfluxDB runs the query against the associated influxdb database
+// there is an implicit assumption here that the first value returned from the iterator
+// will be the value of interest
+func queryInfluxDB(queryAPI api.QueryAPI, query string) (float64, error) {
+	result, err := queryAPI.Query(context.Background(), query)
+	if err != nil {
+		return 0, err
+	}
+
+	for result.Next() {
+		// Do one iteration to get the value
+		val := result.Record().Value()
+		if assertedVal, ok := val.(float64); ok {
+			influxDBLog.Info("the value was this")
+			return assertedVal, nil
+		}
+		return 0, fmt.Errorf("error parsing queried value to float")
+	}
+
+	return 0, fmt.Errorf("no results")
+}
+
+// GetMetrics connects to influxdb via the client and returns a value based on the query
+func (s *influxDBScaler) GetMetrics(ctx context.Context, metricName string, metricSelector labels.Selector) ([]external_metrics.ExternalMetricValue, error) {
+	// Grab QueryAPI to make queries to influxdb instance
+	queryAPI := s.client.QueryAPI(s.metadata.organizationName)
+
+	value, err := queryInfluxDB(queryAPI, s.metadata.query)
+	if err != nil {
+		return []external_metrics.ExternalMetricValue{}, err
+	}
+
+	metric := external_metrics.ExternalMetricValue{
+		MetricName: metricName,
+		Value:      *resource.NewQuantity(int64(value), resource.DecimalSI),
+		Timestamp:  metav1.Now(),
+	}
+
+	return append([]external_metrics.ExternalMetricValue{}, metric), nil
+}
+
+// GetMetricSpecForScaling returns the metric spec for the Horizontal Pod Autoscaler
+func (s *influxDBScaler) GetMetricSpecForScaling() []v2beta2.MetricSpec {
+	targetMetricValue := resource.NewQuantity(int64(s.metadata.thresholdValue), resource.DecimalSI)
+	externalMetric := &v2beta2.ExternalMetricSource{
+		Metric: v2beta2.MetricIdentifier{
+			Name: kedautil.NormalizeString(fmt.Sprintf("%s-%s", "influxdb", s.metadata.organizationName)),
```

Comment:
The metric name should be unique enough, to allow specify multiple triggers in one Scaler and it should be possible to distinguish between those individual metric names.

I wonder wheter is `s.metadata.organizationName` enough?

Reply:
I think that could be enough.

Reply:
What if I want to create two (or more) triggers that use the same organization name? I am not an InfluxDB expert, but I suppose this could be possible?

Reply:
@zroubalik Yes that definitely could be possible

Reply:
So in this case (when I define two or more influxdb scalers in ScaledObject, that will use the same organization) the generated metric name will be same, right? And that's not ideal :)

Reply:
@zroubalik Agreed. There is no passed in info that would make this name unique enough. Would using a timestamp suffice?